GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java

```diff
@@ -716,6 +723,32 @@ private synchronized int writeInstanceData(Sheet sheet, final int startRow,
 
         return maxRow + 1;
     }
+    
+    private String constructDisplayName(
+            List<QuestionDto> questionList,
+            InstanceData instanceData
+            ) {
+        StringBuilder dname = new StringBuilder();
+        
+        for (QuestionDto q : questionList) {
+            if (q != null && q.getKeyId() != null && q.getType() != null) {
```

Comment:
Perhaps we can make these checks (`null`, `keyId`, `type`) at the point in time when we add the display name question to `displayNamePartList `.  It is something that only happens at the beginning of the report generation. That way we don't have an extra `if` block here